modules/Microsoft.ContainerInstance/containerGroups/.test/encr/dependencies.bicep

```
@@ -0,0 +1,61 @@
+@description('Required. The name of the managed identity to create.')
+param managedIdentityName string
+
+@description('Optional. The location to deploy resources to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Key Vault to create.')
+@minLength(3)
+@maxLength(24)
+param keyVaultName string
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+  name: managedIdentityName
+  location: location
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: keyVaultName
+  location: location
+  properties: {
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: tenant().tenantId
+    enablePurgeProtection: true // Required by batch account
+    softDeleteRetentionInDays: 7
+    enabledForTemplateDeployment: true
+    enabledForDiskEncryption: true
+    enabledForDeployment: true
+    enableRbacAuthorization: true
+    accessPolicies: []
+  }
+
+  resource key 'keys@2022-07-01' = {
+    name: 'keyEncryptionKey'
+    properties: {
+      kty: 'RSA'
+    }
+  }
+}
+
+// Ref: https://learn.microsoft.com/en-us/azure/container-instances/container-instances-encrypt-data#create-service-principal-for-aci
+resource keyPermissions 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  name: guid('msi-${keyVault::key.id}-${location}-Azure Container Instance Service-Key Vault Crypto User')
+  scope: keyVault
+  properties: {
+    principalId: '8b659b68-1eb9-4ea5-ab00-a6a182520436' // 'Azure Container Instance Service' Service Principal Object Id
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'e147488a-f6f5-4113-8e2d-b22465e65bf6') // Key Vault Crypto Service Encryption User. Allows Keys: get, list, wrap key, unwrap key
```

I'd suggest to limit the comment to the role name, like in other test files
```suggestion
    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'e147488a-f6f5-4113-8e2d-b22465e65bf6') // Key Vault Crypto Service Encryption User.
```

---

I added it explicitely as the error message would point me to it and I'd consider it useful information as the role itself doesn't tell you much actually (even if you search for it). Why would you consider additional context to be too much?

---

I just wouldn't see the need of documenting a built-in role for which allowed operations are known (e.g. https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-guide?tabs=azure-cli#azure-built-in-roles-for-key-vault-data-plane-operations). We do not do the same for other module tests

---

Right we don't - in this case I only added it because it was explicitely called out and I figured it wouldn't be only interesting to me (as this also explains what you'd need for access policies). Also, I'm not sure why we're discussing this - the implementation is 1:1 like in all other implementations and we're just debating the richness of a comment? 

Don't get me wrong, if you think it's a really bad idea to have this 'extra' content (for example, as 'search & replace' would be more complicated) I'm happy to remove it. If not, I just can't get over the fact that we're debating a sidenote in a comment :D

---

I'd say there is an undefined wiggle room :P Let's resolve this like adults and pull somebody else in to tip the scale :D

@ahmadabdalla @MariusStorhaug @rahalan @ChrisSidebotham @elanzel  you're on! :P Who answers first, counts.

---

I vote for keeping it as per Alex's suggestion. **or** replacing it with the statement from the [documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#key-vault-crypto-service-encryption-user). No harm for having clarity.

_Read metadata of keys and perform wrap/unwrap operations_

---

I yield!